### PR TITLE
Svelte CreateQuery tests. 

### PR DESCRIPTION
**Pull request by rivo420**
- Added CreateQuery type tests.
- Changed Svelte's accessors flag to 'true' to allow prop access from tests.
- Made svelte-check run before vitest to check types.

### CODE REVIEW

**Comment by TkDodo** [comment]
@lachlancollins FYI

**Comment by rivo420** [comment]
(bug): onSuccess isn't being called on refetch().

File: packages/svelte-query/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "build": "svelte-kit sync && svelte-package && rimraf ./build/lib/package.json",
     "check": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json",
     "test:eslint": "svelte-kit sync && ../../node_modules/.bin/eslint --ext .svelte,.ts ./src",
-    "test:jest": "svelte-kit sync && vitest run --coverage"
+    "test:jest": "npm run check && vitest run --coverage"
```
Comment (lachlancollins) [inline packages/svelte-query/package.json:21]: I've restructured the testing scripts here so this isn't needed: https://github.com/TanStack/query/pull/4779

File: packages/svelte-query/src/__tests__/createQuery/CreateQuery.test.svelte [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <script lang="ts">
-  import { createQuery, QueryClient } from '../lib'
-  import { setQueryClientContext } from '../lib/context'
+  import { createQuery, QueryClient } from '$lib'
+  import { setQueryClientContext } from '$lib/context'
```
Comment (lachlancollins) [inline packages/svelte-query/src/__tests__/createQuery/CreateQuery.test.svelte:3]: I'm preparing to remove sveltekit and just use vite-plugin-svelte, can you switch this back to ../lib imports?

**Comment by lachlancollins** [comment]
Huge, great work @rivo420 - I'll have some time to look into the bug tomorrow or Wednesday, would you be able to create an issue and a repro?

**Comment by lachlancollins** [comment]
@rivo420 could you merge with main? I've slightly restructured the src folder which will break your imports

**Comment by lachlancollins** [comment]
Hi @rivo420 , I've tested refetch() in the `star-wars` example and it's definitely called onSuccess. Could there be an issue with the test? Nothing obvious stands out to me.

**Comment by lachlancollins** [comment]
Hi @rivo420 , are you still available to work on this PR? I'd greatly appreciate the tests for the v5 work.

**Pull request closed without merging**